src/zenml/models/v2/core/trigger.py

```diff
@@ -0,0 +1,114 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Collection of all models concerning triggers."""
+from datetime import datetime
+from uuid import UUID
+
+from pydantic import Field
+
+from zenml.constants import STR_FIELD_MAX_LENGTH
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedFilter,
+    WorkspaceScopedRequest,
+    WorkspaceScopedResponse,
+    WorkspaceScopedResponseBody,
+    WorkspaceScopedResponseMetadata,
+)
+from zenml.models.v2.base.update import update_model
+from zenml.models.v2.core.action_plan import ActionPlanResponse
+from zenml.models.v2.core.event_filter import EventFilterResponse
+
+
+# ------------------ Request Model ------------------
+class TriggerRequest(WorkspaceScopedRequest):
+    """Model for creating a new Trigger."""
+
+    name: str = Field(
+        title="The name of the Trigger.", max_length=STR_FIELD_MAX_LENGTH
+    )
+    description: str = Field(
+        default="",
+        title="The description of the trigger",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    event_filter_id: UUID = Field(title="The id of the event filter.")
+    action_plan_id: UUID = Field(title="The id of the action plan.")
```

Comment:
Are the filter and action plan reusable ? From a business logic perspective, would we ever need to configure these separately and then group them together under the same trigger ?

If not, then you should model this more as [an aggregation/composition instead of an association](https://www.visual-paradigm.com/guide/uml-unified-modeling-language/uml-aggregation-vs-composition/), meaning that you can directly include here all the fields that are part of the filter/source classes. This keeps the domain models and even the database more compact and the implementation less comples.

Reply:
very good point - would you then just serialize/deserialize these into database columns on the trigger?